cpu/stm32_common/periph/pm.c

```diff
@@ -49,13 +61,21 @@ void pm_set(unsigned mode)
             /* Clear PDDS and LPDS bits to enter stop mode on */
             /* deepsleep with voltage regulator on */
             PWR->CR &= ~(PWR_CR_PDDS | PWR_CR_LPDS);
+            PWR->CR |= PM_STOP_CONFIG;
             /* Set SLEEPDEEP bit of system control block */
             deep = 1;
             break;
     }
 #endif
 
     cortexm_sleep(deep);
+
+#if defined(CPU_FAM_STM32F1) || defined(CPU_FAM_STM32F2) || defined(CPU_FAM_STM32F4)
```

Comment:
I've discussed this offline with @haukepetersen, and the latter solution (replace __disable_irq/__enable_irq with irq_disable/irq_restore) seems the most elegant to us. It only changes these two lines, and it should work with both boards using pm_layered and with boards that don't.

@vincent-d would you mind updating the PR accordingly?